src/main/java/uk/gov/hmcts/ccd/endpoint/exceptions/ReferenceKeyConcurrencyException.java

```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.ccd.endpoint.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.CONFLICT)
+public class ReferenceKeyConcurrencyException extends ApiException {
```

Comment:
Could we change the name ReferenceKeyConcurrencyException to something like, DuplicateReferenceKeyException? There doesn't have to be a concurrency matter involved in the case. So, concurrency word may be irrelevant.

Reply:
Changed to ReferenceKeyUniqueConstraintException